### DISCUSSION TITLE
Jaxb names customization

### DISCUSSION THREAD

**Discussion by theseeker58** [victools/jsonschema-generator#462] [Q&A] [answered]
I'm using xjc for automatic generation of java classes hierarchy from a xsd issued by ISO (International Organization for Standardization).
I have also used an external xjb file to customize classes and properties names. I wanted to avoid classes named GrpHdr (the xml tag) instead of GroupHeader or Ccy instead of Currency.
Then I register the JakartaXmlBindAnnotationModule to allow Jackson to deserialize json bodies perfectly aligned to xml tag names.
Unfortunately Victools generates the json schema accordingly to classes and properties names customized through xjb (GroupHeader and not GrpHdr, in the example above).
The alternatives currenly could be:
1) not care about elegant coding and accept ugly signatures like getGrpHdr() or getCcy()
2) customize classes generation through xjb injecting @JsonProperty annotations prefectly aligned to XML tags (quite boring though, and I need to check first if it works)

Do you know if is there in Victools a smarter way to customize json schema generation using @XMLElement jaxb annotation already present in jaxb classes?

```java
    @XmlElement(name = "GrpHdr", required = true)
    @JacksonXmlProperty(namespace = "urn:iso:std:iso:20022:tech:xsd:pain.013.001.10")
    @JsonProperty(required = true)
    @NotNull
    @Valid
    protected GroupHeader groupHeader;
```

Many thanks

**Comment by CarstenWickner** [comment, marked as answer]
Hi @theseeker58,

You correctly referred to Jackson's `@JsonProperty` annotations already being supported, which tells you it can be configured. 😃
That's what "Property Name Overrides" are for (as I call them). They are mentioned in the [documentation](https://victools.github.io/jsonschema-generator/#names-of-fields-methods-in-an-object-39-s-properties).

The example given there only needs a tiny adjustment to also apply for `@XmlElement` annotations:
```java
configBuilder.forFields()
    .withPropertyNameOverrideResolver(field -> Optional
            .ofNullable(field.getAnnotationConsideringFieldAndGetter(XmlElement.class))
            .map(XmlElement::name).orElse(null));
```

That should do it.

**Comment by theseeker58** [comment]
Thank you @CarstenWickner  you made my day!
At first just reading the generated schema apparently everything worked fine, but when the schema validation threw errors I realized that I had to resolve XmlAttributes as well.
So added

```
   configBuilder.forFields()
        .withPropertyNameOverrideResolver(field -> Optional
             .ofNullable(field.getAnnotationConsideringFieldAndGetter(XmlAttribute.class))
                .map(XmlAttribute::name).orElse(null));

```
For the record I'm preparing a benchmark to compare json schema validation vs. bean validation under heavy load even though I guess that, in a reactive approach caching the body vs. streaming can be penalizing. On the contrary json schema validation makes useless jaxb classes and the body can be directly mapped to avro classes (generated automatically from jaxb classes too) for subsequent kafka streaming. Considering that all stuff is automatically generated from the original xsd scheme it makes sense.

Thank you again
